### PR TITLE
Disable selection if inside block

### PR DESCRIPTION
**Pull request by MoMMde**
# Update
I disabled the Block Selection when you are inside a Block

# Config
It is configurable but i don't think it will be needed in future

# Example

### CODE REVIEW

**Comment by MoMMde** [comment]
Fluid blocks not working

**Comment by MoMMde** [comment]
Flying Underground currently has some issues

**Comment by MoMMde** [comment]
Please don't ask why this takes soooo long

File: src/main/java/meteordevelopment/meteorclient/systems/modules/render/BlockSelection.java [hunk 1]
```diff
@@ -65,6 +76,15 @@ public BlockSelection() {
     private void onRender(Render3DEvent event) {
         if (mc.crosshairTarget == null || !(mc.crosshairTarget instanceof BlockHitResult result)) return;
 
+        BlockPos playerEyePos = new BlockPos(mc.player.getEyePos());
+        int crossX = (int) Math.floor(mc.crosshairTarget.getPos().x);
+        int crossY = (int) Math.floor(mc.crosshairTarget.getPos().y);
+        int crossZ = (int) Math.floor(mc.crosshairTarget.getPos().z);
```
Comment (MineGame159) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/render/BlockSelection.java:82]: Why are you calculating all of this when it's not needed?

File: src/main/java/meteordevelopment/meteorclient/systems/modules/render/BlockSelection.java [hunk 2]
```diff
@@ -65,6 +76,15 @@ public BlockSelection() {
...
+
+        if(hideInside.get() && result.isInsideBlock()) {
```
Comment (MineGame159) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/render/BlockSelection.java:84]: Can you follow our standards and put a space between `if` and `(`. Also the return could be on the same line but that's not important.

File: src/main/java/meteordevelopment/meteorclient/systems/modules/render/BlockSelection.java [hunk 3]
```diff
@@ -17,8 +18,11 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Box;
 import net.minecraft.util.math.Direction;
+import net.minecraft.util.math.Vec3d;
 import net.minecraft.util.shape.VoxelShape;
 
+import java.math.RoundingMode;
```
Comment (MineGame159) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/render/BlockSelection.java:24]: Why are you adding imports that are never used? There is another one at the top.

**Review by MineGame159 — CHANGES_REQUESTED** [review]

File: src/main/java/meteordevelopment/meteorclient/systems/modules/render/BlockSelection.java (hunk 3 above)
Comment (MoMMde) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/render/BlockSelection.java:24]: Forgot to remove the stuff

Comment (MoMMde) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/render/BlockSelection.java:24]: I'm new to Mod Development and i needed to figure out what each field does and i just started messing around with different fields i got and in the end i found my current solution but now forgot to remove the old things

**Pull request merged**